packages/lib/entityPermissionUtils.server.ts

```diff
@@ -1,125 +1,73 @@
+import { PermissionString } from "@calcom/features/pbac/domain/types/permission-registry";
+import { PermissionCheckService } from "@calcom/features/pbac/services/permission-check.service";
 import { MembershipRole } from "@calcom/prisma/enums";
 
-export const enum ENTITY_PERMISSION_LEVEL {
-  NONE,
-  // It is owned by user and user has write access to it
-  USER_ONLY_WRITE,
-  // All members of the team has access to it and user has read access to it
-  TEAM_READ_ONLY,
-  // All members of the team has access to it and user has write access to it
-  TEAM_WRITE,
-}
+export type EntityPermission = {
+  userId: number | null;
+  teamId: number | null;
+};
 
 export async function canEditEntity(
-  entity: Parameters<typeof getEntityPermissionLevel>[0],
-  userId: Parameters<typeof getEntityPermissionLevel>[1]
+  entity: EntityPermission,
+  userId: number,
+  permission: PermissionString = "routingForm.update"
 ) {
-  const permissionLevel = await getEntityPermissionLevel(entity, userId);
-  return (
-    permissionLevel === ENTITY_PERMISSION_LEVEL.TEAM_WRITE ||
-    permissionLevel === ENTITY_PERMISSION_LEVEL.USER_ONLY_WRITE
-  );
-}
+  if (entity.teamId) {
+    const permissionService = new PermissionCheckService();
+    const hasEditPermission = await permissionService.checkPermission({
+      teamId: entity.teamId,
+      userId,
+      permission,
+      fallbackRoles: [MembershipRole.ADMIN, MembershipRole.OWNER],
+    });
```

Comment:
Now simpily check if the user has permission on this teamId -> if pbac is disabled we use fallbackRoles